### PR TITLE
runner: full log dump after "-l debug" option

### PR DESCRIPTION
**Pull request by niewim19**
JIRA: CI-107

## Description

Add functionality to "-l debug" option of runner so that it dumps everything that was read into a file in /tmp/ directory

## Motivation and Context

For inspection purposes runner should have an option to dump everything it has read from target device.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by agkaminski** [comment]
Works for me :+1:

File: trunner/device.py [hunk 1]
```diff
@@ -18,16 +18,16 @@
 
 class RunnerFactory:
     @staticmethod
-    def create(target, serial):
+    def create(target, serial, log):
```
Comment (damianloew) [inline trunner/device.py:21]: Maybe 

```suggestion
    def create(target, serial, log=False):
```

**Review by damianloew — COMMENTED** [review]
I think it could be helpful, but I prefer to test it manually on all available targets

**Review by damianloew — APPROVED** [review]
Works properly on RT platform too. LGTM

**Review by damianloew — APPROVED** [review]

**Pull request merged**